pkg/gds/admin.go

```diff
@@ -869,17 +867,10 @@ func (s *Admin) prepareVASPDetail(vasp *pb.VASP, log zerolog.Logger) (out *admin
 	// Must be done after verified contacts is computed
 	// WARNING: This is safe because nothing is saved back to the database!
 	vasp.Extra = nil
-	if vasp.Contacts.Administrative != nil {
-		vasp.Contacts.Administrative.Extra = nil
-	}
-	if vasp.Contacts.Legal != nil {
-		vasp.Contacts.Legal.Extra = nil
-	}
-	if vasp.Contacts.Technical != nil {
-		vasp.Contacts.Technical.Extra = nil
-	}
-	if vasp.Contacts.Billing != nil {
-		vasp.Contacts.Billing.Extra = nil
+	iter := models.NewContactIterator(vasp.Contacts, false, false)
+	for iter.Next() {
+		contact, _, _ := iter.Value()
+		contact.Extra = nil
```

Comment:
This is so much nicer! Sorry - I'm very happy about these changes, it will make things so much easier!